clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -6056,8 +6056,21 @@ bool clang::CodeGen::noSystemDebugInfo(const Decl *D,
     if (isa<TypedefDecl>(D))
       return false;
 
-    // -fno-system-debug was used.  Do not generate debug info.
-    if (CGM.getCodeGenOpts().NoSystemDebug)
+#ifndef WINDOWS
+    // Generate debug for _M_insert_aux to allow debugging for
+    // deque's value_type.
+    // _M_insert_aux is not used by Windows implementation, so
+    // there is no need for this check on Windows.
+    const auto *Constructor = dyn_cast<CXXConstructorDecl>(D);
+    if (Constructor && Constructor->getNameAsString() == "deque")
+      return false;
+    const auto *Method = dyn_cast<CXXMethodDecl>(D);
+    if (Method && Method->getNameAsString() == "_M_insert_aux")
+      return false;
```

Comment:
What is unique about these two entries that they are special-cased?  Feels very hacky.

Reply:
Remove special case to preserve _M_insert_aux.  Try to find clean solution later.  Small loss of debuggability can be dealt with by using -fsystem-debug.